### ISSUE TITLE
ONNX inference example or guidance?

### ISSUE THREAD

**Issue by chussboi96** [RangiLyu/nanodet#585]
Hello,

I trained a NanoDet-Plus model (input size 416, reg_max=7, 1 class) and exported to ONNX. 
The model runs in ONNXRuntime and returns output of shape (1,3549,33).

Questions:
- What is the exact layout of the ONNX outputs? (Currently assuming cls_logits + bbox_distrib).
- Is there an official ONNX inference script (`demo_onnx.py`) or guidance for decoding and NMS?
- Should preprocessing use simple resize or letterbox as in training?

I tried decoding with softmax + DFL + NMS, but I get either no detections or incorrect results.

Config: 
"""
save_dir: workspace/nanodet-plus-m-416-barcode

class_names:
  - barcode

model:
  arch:
    name: OneStageDetector
    backbone:
      name: ShuffleNetV2
      model_size: 1.5x
      out_stages: [2, 3, 4]
      activation: HardSwish
    fpn:
      name: GhostPAN
      in_channels: [176, 352, 704]
      out_channels: 96
      kernel_size: 5
      use_depthwise: true
      activation: HardSwish
    head:
      name: NanoDetPlusHead
      num_classes: 1
      input_channel: 96
      feat_channels: 96
      stacked_convs: 2
      strides: [8, 16, 32]
      reg_max: 7
      activation: HardSwish
      assigner:
        name: DynamicSoftLabelAssigner
        topk: 13
      loss:
        loss_qfl:
          name: QualityFocalLoss
          use_sigmoid: true
          beta: 2.0
          loss_weight: 1.0
        loss_dfl:
          name: DistributionFocalLoss
          loss_weight: 0.25
        loss_bbox:
          name: GIoULoss
          loss_weight: 2.0

data:
  train:
    name: CocoDataset
    img_path: D:/merged_dataset/train
    ann_path: D:/merged_dataset/train/_annotations.coco.json
    input_size: [416, 416]
    keep_ratio: true
    pipeline:
      flip: 0.5
      crop:
        crop_ratio: [0.9, 1.0]
        prob: 0.5
      color_jitter:
        brightness: 0.4
        contrast: 0.4
        saturation: 0.4
        hue: 0.1
        prob: 0.5
      affine:
        degrees: 20
        translate: 0.1
        scale: [0.8, 1.2]
        shear: 10
        prob: 0.5
      mixup: 0.1
      resize:
        target_size: [416, 416]
        keep_ratio: true
      pad:
        size: [416, 416]
        fill_value: 0
      normalize: [[103.53, 116.28, 123.675], [57.375, 57.12, 58.395]]
  val:
    name: CocoDataset
    img_path: D:/merged_dataset/val
    ann_path: D:/merged_dataset/val/_annotations.coco.json
    input_size: [416, 416]
    keep_ratio: true
    pipeline:
      resize:
        target_size: [416, 416]
        keep_ratio: true
      pad:
        size: [416, 416]
        fill_value: 0
      normalize: [[103.53, 116.28, 123.675], [57.375, 57.12, 58.395]]
  test:
    name: CocoDataset
    img_path: D:/merged_dataset/test
    ann_path: D:/merged_dataset/test/_annotations.coco.json
    input_size: [416, 416]
    keep_ratio: true
    pipeline:
      resize:
        target_size: [416, 416]
        keep_ratio: true
      pad:
        size: [416, 416]
        fill_value: 0
      normalize: [[103.53, 116.28, 123.675], [57.375, 57.12, 58.395]]

device:
  gpu_ids: [0]
  workers_per_gpu: 4
  batchsize_per_gpu: 64
  precision: 16

schedule:
  load_model: C:/Users/LOQ/Downloads/nanodet-plus-m_416_checkpoint.ckpt
  optimizer:
    name: AdamW
    lr: 0.004
    weight_decay: 0.05
  warmup:
    name: linear
    steps: 500
    ratio: 0.00066667
  total_epochs: 50
  lr_schedule:
    name: CosineAnnealingLR
    T_max: 50
  val_intervals: 5
  save_checkpoints: true
  checkpoint_dir: workspace/nanodet-plus-m-416-barcode

log:
  interval: 50

evaluator:
  name: CocoDetectionEvaluator
  save_key: mAP
  iou_thrs: [0.5]
"""
ONNX output shape: `(1,3549,33)`
Environment: PyTorch 1.x, ONNXRuntime, Windows 11.

Would you be able to provide an example script for ONNX inference, or confirm the expected decode pipeline?

Thank you!
